lib/improver/percentile.py

```diff
@@ -0,0 +1,149 @@
+
+    """
+
+    # Default percentile boundaries to calculate at.
+    DEFAULT_PERCENTILES = [0, 5, 10, 20, 25, 30, 40, 50,
+                           60, 70, 75, 80, 90, 95, 100]
+
+    def __init__(self, collapse_coord, percentiles=None):
+        """Create a PDF plugin with a given source plugin.
+
+        Parameters
+        ----------
+        collapse_coord : str or iris.coord.DimCoord (or list of either)
+            The name or coordinate definition of the coordinate(s) to collapse
+            over.
+
+        percentiles : list (optional)
+            Percentile values at which to calculate; if not provided uses
+            DEFAULT_PERCENTILES.
+
+        """
+        if not isinstance(collapse_coord, list):
+            collapse_coord = [collapse_coord]
+        for coord_item in collapse_coord:
+            _type_test(coord_item)
+
+        if percentiles is not None:
+            self.percentiles = [int(value) for value in percentiles]
+        else:
+            self.percentiles = self.DEFAULT_PERCENTILES
+
+        # Collapsing multiple coordinates results in a new percentile
+        # coordinate, its name suffixed by the original coordinate names. Such
+        # a collapse is cummutative (i.e. coordinate order doesn't matter).
+        # However the coordinates are sorted such that the resulting
+        # percentile coordinate has a consistent name regardless of the order
+        # in which the user provides the original coordinate names.
+        self.collapse_coord = sorted(collapse_coord, _sort_coord)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        desc = ('<PercentileConverter: collapse_coord={}, percentiles={}'
+                .format(self.collapse_coord, self.percentiles))
+        return desc
+
+    def process(self, cube):
+        """Create a cube containing the percentiles as a new dimension.
+
+        What's generated is:
+            * 13 percentiles - (5%, 10%, 20%, 25%, 30%, 40%, 50%, 60%,
+              70%, 75%, 80%, 90%, 95%)
+
+        Parameters
+        ----------
+        cube : iris.cube.Cube instance
+            Given the collapse coordinate, convert the set of values
+            along that coordinate into a PDF and extract percentiles
+            and min, max, mean, stdev.
+
+        Returns
+        -------
+        cube : iris.cube.Cube instance
+            A single merged cube of all the cubes produced by each percentile
+            collapse.
+
+        """
+        # Store data type and enforce the same type on return.
+        data_type = cube.dtype
+
+        try:
+            result = cube.collapsed(self.collapse_coord,
+                                    iris.analysis.PERCENTILE,
+                                    percent=self.percentiles)
+            result.data = result.data.astype(data_type)
+            return result
+        except:
+            raise CoordinateNotFoundError(
```

Comment:
I've made this test more explicit. I don't expect any other failures, so I've removed the try/except. If it falls over the iris error message should indicate why.